src/plugins/ui/selection_inputs_manager.ts

```diff
@@ -51,17 +51,38 @@ export class SelectionInputsManagerPlugin extends UIPlugin {
   // ---------------------------------------------------------------------------
 
   allowDispatch(cmd: Command): CommandResult {
+    switch (cmd.type) {
+      case "FOCUS_RANGE":
+      case "CHANGE_RANGE":
+      case "ADD_EMPTY_RANGE":
+      case "REMOVE_RANGE":
+        if (!this.inputs[cmd.id]) {
+          return CommandResult.InvalidInputId;
+        }
+    }
+
     switch (cmd.type) {
       case "FOCUS_RANGE":
         const index = this.currentInput?.getIndex(cmd.rangeId);
         if (this.focusedInputId === cmd.id && this.currentInput?.focusedRangeIndex === index) {
           return CommandResult.InputAlreadyFocused;
         }
         break;
+      case "ADD_EMPTY_RANGE":
+        const input = this.inputs[cmd.id];
+        if (input.inputHasSingleRange && input.ranges.length === 1) {
+          return CommandResult.MaximumRangesReached;
+        }
+        break;
+      case "CHANGE_RANGE": {
+        const input = this.inputs[cmd.id];
+        if (input.inputHasSingleRange && cmd.value.split(",").length > 1) {
+          return CommandResult.MaximumRangesReached;
+        }
+        break;
+      }
     }
-    if (this.currentInput) {
-      return this.currentInput.allowDispatch(cmd);
-    }
```

Comment:
Then we can remove the `allowDispatch` in `selection_input.ts`, right ?

Reply:
100% correct, I amenede the commit in that sense ;-)